sw/device/lib/dif/dif_keymgr.h

```diff
@@ -0,0 +1,683 @@
+ * the handle that must be passed to other functions in this library in each
+ * boot stage. A typical usage of this library during different secure boot
+ * stages is as follows:
+ *
+ * - In Mask ROM:
+ *   - Create a new handle: `dif_keymgr_init()`.
+ *   - Initialize state: `dif_keymgr_initialize_state()`,
+ *   `dif_keymgr_get_status()`.
+ *   - Set state parameters: `dif_keymgr_set_state_params()`.
+ *   - Advance state: `dif_keymgr_advance_state()`, `dif_keymgr_get_status()`.
+ * - In subsequent boot stages, i.e. ROM_EXT, BL0, kernel:
+ *   - Create a new handle: `dif_keymgr_init()`.
+ *   - Generate keys and/or identity seeds:
+ *     `dif_keymgr_generate_versioned_key()`,
+ *     `dif_keymgr_generate_identity_seed()`, `dif_keymgr_get_status()`.
+ *   - Read output: `dif_keymgr_read_output()`.
+ *   - Set state parameters: `dif_keymgr_set_state_params()`.
+ *   - Advance state: `dif_keymgr_advance_state()`, `dif_keymgr_get_status()`.
+ *
+ * @param params Hardware instantiation parameters.
+ * @param[out] keymgr Out-param for the initialized handle.
+ * @return The result of the operation.
+ */
+DIF_WARN_UNUSED_RESULT
+dif_keymgr_result_t dif_keymgr_init(dif_keymgr_params_t params,
+                                    dif_keymgr_t *keymgr);
+
+/**
+ * Initializes key manager after power-on reset.
+ *
+ * After PoR, key manager is in Reset state with a non-deterministic secret
+ * value. This function causes key manager to initialize its secret value using
+ * the random values generated by the entropy source and transition to
+ * Initialized state. Note that Initialized state is not an operational
+ * state and key manager state must be advanced one more time by calling
+ * `dif_keymgr_advance_state()` before keys or identity seeds can be generated.
+ *
+ * This is an asynchronous function because key manager state transitions
+ * involve KMAC operations that can take some time to complete. Clients must
+ * check the status of key manager using `dif_keymgr_get_status()` before
+ * calling other functions in this library.
+ *
+ * The entropy source must be initialized before this function is called because
+ * key manager relies on the entropy source to initialize its secret value.
+ *
+ * This function must be called only once per power cycle.
+ *
+ * @param keymgr A key manager handle.
+ * @return The result of the operation.
+ */
+DIF_WARN_UNUSED_RESULT
+dif_keymgr_result_t dif_keymgr_initialize_state(const dif_keymgr_t *keymgr);
+
+/**
+ * Valid lengths for the `binding_value` array in `dif_keymgr_state_params_t`.
+ */
+typedef enum dif_keymgr_binding_value_len {
+  /**
+   * `binding_value` length when transitioning to `OwnerIntermediateKey` and
+   * `OwnerRootKey` states.
+   */
+  kDifKeymgrBindingValueLenTag = 4,
+  /**
+   * `binding_value` length when transitioning to `CreatorRootKey` state.
+   */
+  kDifKeymgrBindingValueLenRomExtHash = 8,
+} dif_keymgr_binding_value_len_t;
```

Comment:
`RomExtSecurityDescriptor` is currently a 128b value in the tech spec but based on our discussions it is expected to be 256b. We may not need these constants if the software binding value ends up using the same size. Using a union in `dif_keymgr_state_params_t ` instead of this is also another option.

Reply:
`RomExtSecurityDescriptor`is now implemented through software binding (#4486).